chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1276,7 +1276,8 @@ impl EthereumAdapterTrait for EthereumAdapter {
             "block_hash" => call.block_ptr.hash_hex(),
             "block_number" => call.block_ptr.block_number()
         );
-
+        let cache_logger = logger.clone();
+        let cache = cache.clone();
```

Comment:
I would like to move it to more deep scope, but I cannot wrap my head on how to correctly clone it in and_then future.